File: Python/boj/Silver/1912.py
import sys

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    n = int(input())
    numbers = list(map(int, input().split(" ")))
    dp = [0] * n
    dp[0] = numbers[0]

    for i in range(1, n):
        dp[i] = max(dp[i - 1] + numbers[i], numbers[i])

    print(max(dp))

File: JavaScript/boj/Silver/1912.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

const n = +input[0];
const numbers = input[1].split(' ').map(Number);

for (let i = 1; i < n; i++) {
  if (numbers[i - 1] > 0) numbers[i] += numbers[i - 1];
}

console.log(Math.max(...numbers));
